apps/src/templates/sessions/LockoutPanel.jsx

```diff
@@ -0,0 +1,283 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import Button from '../Button';
+import i18n from '@cdo/locale';
+import {getStore} from '../../redux';
+import {isEmail} from '@cdo/apps/util/formatValidation';
+import cookies from 'js-cookie';
+import * as color from '../../util/color';
+import headerImage from './images/lockout_penguin.png';
+import headerThanksImage from './images/dancing_penguin.png';
+
+/**
+ * This panel represents the page that is displayed to accounts that are being
+ * locked from accessing certain parts of the site until they get parental
+ * permission. This panel gives a form to request (or see details about a
+ * pending request for) parental permission.
+ */
+class LockoutPanel extends Component {
```

Comment:
Can we make a new React component a functional component instead of a class component? I'm happy to pair with you to shift around the syntax or twist it around myself if that's easier.

Reply:
Do you have an example of a functional component that is similar to our case here somewhere in our codebase to use to conform to convention?

Reply:
I'd particularly like an example of a component with form interactivity... that's the challenge I see in re-wrapping it in the functional style.

Reply:
Here's an example in our codebase (I think it has a lot of interactions) -- https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/music/views/ChordPanel.tsx

Hooks are weird (I think useState is the only semi-intuitive one, but the example component has a couple different hooks in it). But the React.dev notes on hooks are good - https://react.dev/reference/react

Reply:
This is helpful! Good to see the conventions to follow... I already did the work last night... pretty straightforward. Much prefer the "functional" style.

Reply:
Done. Ish. Do I need a `useCallback` if there's no state being used? It's just reacting to input events. It's always the same function.

Global redux state can't be wrapped very effectively, I don't think... but the things being pulled from that are the `isRTL` bool which can only be changed via changing the locale which is always a page refresh anyway.

Reply:
I don't think you need a `useCallback` and yes, I agree on the redux store and `isRTL`